bin/commands/internal/start/component/index.ts

```diff
@@ -93,8 +95,27 @@ export function execute(componentId: string, runInBackground: boolean=false) {
         } else {
           // wait for all background subprocesses created by bin/start.sh exit
           // re-source libs is necessary to reclaim shell functions since this will be executed in a new shell
-          //TODO does this do the same as the shell script before it?
-          shell.execSync('sh', '-c', `cd ${COMPONENT_DIR} ; cat "${fullPath}" | { echo ". \"${ZOWE_CONFIG.zowe.runtimeDirectory}/bin/libs/configmgr-index.sh\"" ; cat ; echo; echo wait; } | /bin/sh`);
+          const startScriptContents = `cd ${COMPONENT_DIR} ; . "${ZOWE_CONFIG.zowe.runtimeDirectory}/bin/libs/configmgr-index.sh" ; ${xplatform.loadFileUTF8(fullPath, xplatform.AUTO_DETECT)} ; wait;`;
+          const pipeArray = os.pipe();
+          if (!pipeArray) {
+            //TODO error message
+            common.printFormattedError("ZWELS", "zwe-internal-start-component", `Error ZWEL???E: Could not create pipe.`);    
+            return;
+          }
+          //TODO this will not work with unicode codepoints longer than a byte
```

Comment:
Since the buffer contains just paths & script content, unless the script is containing things not found in ascii (really unusual aside from .... copyright symbols??) then I guess the only possibility for this to be an issue is if someone has pathnames not found in ASCII.

So I tested how we do with pathnames containing Japanese today, without this PR
PKCS12 gen fails for some reason, but assuming you already have a keyring...
Init passes, prepare passes, and app-server/zss start up.
But all the java servers fail to load with `Error: Could not find or load main class org.springframework.boot.loader.PropertiesLauncher`
So, we have work to do elsewhere too but yeah it looks like the majority of zwe actually does handle unicode alright, so should fix this before merging...

Reply:
IBM recommends to use [POSIX portable character set](https://www.ibm.com/docs/en/zos/3.1.0?topic=system-path-path-name) for paths, otherwise the result is unpredictable. I would not be worried about this.

If the scope of `zwe internal` is still internal, we as a zowe community have under control, what will be in the start-up scripts. I would not be worried about non-ascii chars too.

Reply:
Thats a good point. I guess it's good to go as-is then.